shaarli_client/utils.py

```diff
@@ -24,26 +24,22 @@ def generate_all_endpoints_parsers(subparsers, endpoints):
         generate_endpoint_parser(subparsers, ep_name, ep_metadata)
 
 
-def write_output(args, response):
-    """Write the API response to file or standard output"""
-    if not args.outfile:
-        if args.format == 'json':
-            print(response.json())
-        elif args.format == 'pprint':
-            print(json.dumps(response.json(), sort_keys=True, indent=4))
-        elif args.format == 'text':
-            print(response.text)
+def format_response(output_format, response):
+    """Format the API response to the desired output format"""
+    if output_format == 'json':
+        return response.json()
+    elif output_format == 'pprint':
+        return json.dumps(response.json(), sort_keys=True, indent=4)
+    elif output_format == 'text':
+        return response.text
     else:
-        try:
-            output_file = open(args.outfile, 'w+')
-        except:
-            raise OSError("File %s cannot be written" % args.outfile)
-        if args.format == 'json':
-            output_file.write(response.json())
-        elif args.format == 'pprint':
-            output_file.write(
-                json.dumps(response.json(), sort_keys=True, indent=4)
-            )
-        elif args.format == 'text':
-            output_file.write(response.text)
-        output_file.close()
+        raise ValueError("%s is not a supported format." % output_format)
+
+def output_to_file(output_filename, output):
+    """Write the program output to a file"""
+    try:
+        output_file = open(output_filename, 'w+')
+    except:
```

Comment:
broad catching should be avoided (Pylint might already be complaining on this one ;-) )

here are a couple common ways of catching an exception and re-throwing it with additional information:

```python
# catch, log and re-raise
try:
   something()
except OSError as exc:
    logging.error("my msg %s" % exc.message)
    raise
```

```python
# catch, raise new exception
try:
   something()
except OSError as exc:
    raise OSError("custom msg %s" % exc.message)
```

```python
# catch, reraise from (Python 3)
try:
   something()
except OSError as exc:
    raise OSError("custom msg %s" % exc.message) from exc
```

Reference:
- https://www.python.org/dev/peps/pep-3134/
- https://stackoverflow.com/questions/696047/re-raise-exception-with-a-different-type-and-message-preserving-existing-inform